### DISCUSSION TITLE
Yolo11 vs Yolo26

### DISCUSSION THREAD

**Discussion by pfeatherstone** [ultralytics/ultralytics#23277] [General]
When doing a diff of both architectures, very little has changed in backbone and head. So does it mainly come down to removing DFL and the new optimizer? Yolo11 already had the one shot detector right? I.e. it already removed NMS correct?

<img width="1580" height="833" alt="Screenshot from 2026-01-16 10-16-16" src="https://github.com/user-attachments/assets/daa45542-85d5-4d14-8d29-7b506ba3e543" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @pfeatherstone, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get moving quickly—an Ultralytics engineer will also jump in to assist soon 🙂. In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report (for example, unexpected results when switching between YOLO11 and YOLO26), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
There are changes in loss function and the heads of the models, especially for other tasks like segment (new semantic segmentation loss), OBB (new angle loss) and pose (new RLE loss).

**Reply by pfeatherstone** [reply]
Regarding detection model architecture, it looks like C3k2 takes argument "True" everywhere now. Otherwise it's pretty much unchanged right?

**Reply by pfeatherstone** [reply]
Or does the YAML loader do a lot of massaging when loading a yolo26 model outside of what's defined in the yaml file?

**Reply by glenn-jocher** [reply]
That trailing `True/False` you’re seeing on `C3k2` is the `c3k` flag (it switches the internal block from a plain `Bottleneck` to `C3k`), and when you build from a `.yaml` the model parser will also force `c3k=True` automatically for `m/l/x` scales (`parse_model()` sets `args[3]=True`), so the instantiated graph can differ slightly from what’s written in the YAML ([`parse_model()` reference](https://docs.ultralytics.com/reference/nn/tasks/), [`C3k2` reference](https://docs.ultralytics.com/reference/nn/modules/block/)). Beyond that, it’s mostly the standard depth/width scaling plus passing things like `end2end` into the head, rather than lots of hidden YAML rewriting.

**Reply by pfeatherstone** [reply]
Sweet. What do you reckon were the biggest gains over Yolo11? The architectures look very similar and, correct me if i'm wrong, Yolo11 already had the main Yolo10 innovation with regards to removing NMS. Is the MuonSGD optimizer the biggest killer feature here ?

**Reply by glenn-jocher** [reply]
YOLO26’s biggest wins vs YOLO11 are mainly the **native end-to-end (NMS-free) inference path** plus the simplified box regression (**DFL removal**) and updated training recipe (**ProgLoss+STAL** with **MuSGD**)—MuSGD helps, but it’s not the only driver; also, YOLO11’s standard detection pipeline still uses NMS, so the YOLOv10-style “true end2end” behavior is a YOLO26 addition (and is where a lot of the latency/deployment simplification comes from), with CPU gains called out in the docs (up to ~43% for `yolo26n`)—details in the [YOLO11 vs YOLO26 FAQ](https://docs.ultralytics.com/help/FAQ/) and the [YOLO26 model docs](https://docs.ultralytics.com/models/yolo26/).

**Reply by pfeatherstone** [reply]
I've noticed the activations in SPPF have silently changed. It used to be nn.SILU() for both convs, now the first one has no activation. It's things like that you never see in the YAML configuration file. There are quite a few things  `parse_model()` does behind the scenes.

**Reply by pfeatherstone** [reply]
What's ProgLoss? If you grep the entire repo you only get these results:
```
:~/Downloads/repos/ultralytics$ grep -r "ProgLoss" *
docs/en/models/yolo26.md:- **ProgLoss + STAL**  
docs/en/models/yolo26.md:- **ProgLoss + STAL**: Boosts accuracy, especially on small objects
```
I can't see anything in `v8DetectionLoss` or `E2ELoss` that mentions ProgLoss

**Reply by Y-T-G** [reply]
> What's ProgLoss? If you grep the entire repo you only get these results:
> ```
> :~/Downloads/repos/ultralytics$ grep -r "ProgLoss" *
> docs/en/models/yolo26.md:- **ProgLoss + STAL**  
> docs/en/models/yolo26.md:- **ProgLoss + STAL**: Boosts accuracy, especially on small objects
> ```
> I can't see anything in `v8DetectionLoss` or `E2ELoss` that mentions ProgLoss

@Laughing-q

**Comment by GKXKX123456** [comment]
【Detect task】I trained YOLO11s, YOLO12s, and YOLO26s on my custom dataset. Surprisingly, YOLO11 achieved the highest mAP, followed by YOLO12, with YOLO26 performing the worst. My dataset contains just over 2,000 images with 11 classes. I'm not sure what the issue might be.

**Reply by Y-T-G** [reply]
What's the mAP if you run YOLO26 validation with NMS?

```
from ultralytics import YOLO

model = YOLO("best.pt")
del model.model.model[-1].one2one_cv2

results = model.val(data="data.yaml")
```
